docs(misconf): Add information about selectors

simar7 opened the pull request:
## Description
Add documentation on how to use selectors with policies.

## Related issues
- Close https://github.com/aquasecurity/defsec/issues/1201

itaysk reviewed (commented):
IIUC, we implemented subtype for cloud only but type still supports k8s no?

simar7 commented:
> IIUC, we implemented subtype for cloud only but type still supports k8s no?

That's right

> IIUC, we implemented subtype for cloud only but type still supports k8s no?

That's right. I added more detail on this here https://github.com/aquasecurity/trivy/pull/3703/commits/cc7f5495a4e85cbcfa3eb2fec2fb75fa1bb3b86c

itaysk commented:
@simar7 I'm not sure how that commit helped.
In the table, k8s is missing:
```
| Type                     | Subtype fields required | Example                         | 
|--------------------------|-------------------------|---------------------------------|
| Cloud (AWS, Azure, etc.) | `provider`, `service`   | `provider: aws`, `service: rds` |
```

This statement is incorrect IFAIU
```
Currently, subtypes are only supported with Cloud resources. We plan to add support for all Trivy misconfiguration (Kubernetes and Dockerfiles) types shortly.
```

itaysk commented:
```
For input selectors to work, the following fields are required to be specified per misconfiguration type
```

based on discussion in https://github.com/aquasecurity/defsec/issues/1200 I understand that if selectors are missing the check is broken. So this guidance needs to be more accurate (remove "For input selectors to work")

simar7 commented:
> ```
> For input selectors to work, the following fields are required to be specified per misconfiguration type
> ```
> 

> based on discussion in [aquasecurity/defsec#1200](https://github.com/aquasecurity/defsec/issues/1200) I understand that if selectors are missing the check is broken. So this guidance needs to be more accurate (remove "For input selectors to work")

Actually I've [added](https://github.com/aquasecurity/defsec/pull/1213) the support for no selectors at all. As per our [discussion](https://github.com/aquasecurity/defsec/issues/1200#issuecomment-1445369040), if no selectors are present, the policy should be evaluated for all inputs.  

Regardless, I have updated the documentation to be more explicit. PTAL.

itaysk commented on docs/docs/misconfiguration/custom/selectors.md:
```diff
@@ -0,0 +1,61 @@
+3. Dockerfile `type: dockerfile`
+
+
+The following selectors also support the corresponding subtypes:
```
since subtypes relate to types, I would document them in the same list/table.

itaysk commented on docs/docs/misconfiguration/custom/selectors.md:
```diff
@@ -0,0 +1,61 @@
+
+1. Cloud `type: cloud`
+2. Kubernetes `type: kubernetes`
+3. Dockerfile `type: dockerfile`
```
also none as per your last comment

itaysk commented on docs/docs/misconfiguration/custom/selectors.md:
```diff
@@ -0,0 +1,61 @@
+| Cloud (AWS, Azure, etc.) | `provider`, `service`   | `provider: aws`, `service: rds` |
+
+
+You can also define more than one subtype if your policy applies to multiple resources, in this case the policy
```
also multiple selectors?

itaysk commented on docs/docs/misconfiguration/custom/selectors.md:
```diff
@@ -0,0 +1,61 @@
+You can also define more than one subtype if your policy applies to multiple resources, in this case the policy
+will be evaluated for either of the subtypes.
+
+Currently, subtypes are only supported with Cloud resources. We plan to add support for all Trivy misconfiguration (Kubernetes and Dockerfiles) types shortly.
```
this is redundant IMO given the table above

itaysk reviewed (approved):
LGTM with minor comments

knqyf263 commented:
Itay's comments are not addressed yet. Should we wait to merge until they're fixed or do we want to merge this PR now?

simar7 commented:
> Itay's comments are not addressed yet. Should we wait to merge until they're fixed or do we want to merge this PR now?

Sorry I missed this PR, I pushed some updates based on Itay's feedback.
